### PR TITLE
Bug fix in parse_markdown

### PR DESCRIPTION
**Pull request by klausfrieler**
See my issue.

### CODE REVIEW

File: R/i18n.R [hunk 1]
```diff
@@ -234,9 +234,12 @@ i18n_check <- function(x) {
   if (length(x$sub) > 0 && is.null(names(x$sub)))
     stop("argument <sub> for function i18n_check() was missing names, ",
          "instead got ", utils::capture.output(print(x$sub)))
-  if (!is.scalar.character(x$x))
+  if (!is.scalar.character(x$x)){
+    browser()
```
Comment (pmcharrison) [inline R/i18n.R:238]: Needs removing

File: R/test-elements.R [hunk 2]
```diff
@@ -1093,7 +1093,7 @@ checkbox_page <-
     )
     get_answer <- function(input, ...)
       if (is.null(input[[label]])) {
-        character(0)
+        ""
```
Comment (pmcharrison) [inline R/test-elements.R:1096]: Are you sure this is what we want?

File: R/test-elements.R [hunk 3]
```diff
@@ -1445,7 +1445,10 @@ loop_while <- function(test, logic) {
                 session = session, opt = opt)
     if (!is.scalar.logical(res)) stop("<test> did not return a ",
                                       "logical scalar")
-    if (res) skip_n_pages(state, - (n + 1L))
+    if (res){
+      message(sprintf("loop_while: Skipping %d pages", n))
+      skip_n_pages(state, - (n + 1L))
```
Comment (pmcharrison) [inline R/test-elements.R:1450]: Do you want to keep this in, or can it be removed now?

File: R/test-elements.R [hunk 4]
```diff
@@ -1495,7 +1498,10 @@ while_loop <- function(test, logic) {
       if (!is.scalar.logical(res)) stop("<test> did not return a ",
                                         "logical scalar")
       if ((skip_when == "pass" && res) ||
-          (skip_when == "fail" && !res)) skip_n_pages(state, skip_len)
+          (skip_when == "fail" && !res)){
+        message(sprintf("while_loop: Skipping %d pages (skip_when = %s)", skip_len, skip_when))
```
Comment (pmcharrison) [inline R/test-elements.R:1502]: Do you want to keep this in, or can it be removed now?

File: R/test-elements.R [hunk 5]
```diff
@@ -1557,7 +1563,10 @@ conditional <- function(test, logic) {
                 session = session, opt = opt)
     if (!is.scalar.logical(res)) stop("<test> did not return a ",
                                       "logical scalar")
-    if (!res) skip_n_pages(state, n)
+    if (!res) {
+      message(sprintf("Conditional: Skipping %d pages", n))
```
Comment (pmcharrison) [inline R/test-elements.R:1567]: Do you want to keep this in, or can it be removed now?

**Review by pmcharrison — COMMENTED** [review]
I updated the CI scripts on master to fix a build problem. Would you mind merging master into your branch again, so we can see if we can make the tests pass?

File: R/test-elements.R (hunk 5 above)
Comment (klausfrieler) [reply R/test-elements.R:1567]: Can be removed

File: R/test-elements.R (hunk 4 above)
Comment (klausfrieler) [reply R/test-elements.R:1502]: Can be removed

File: R/test-elements.R (hunk 3 above)
Comment (klausfrieler) [reply R/test-elements.R:1450]: Can be removed

File: R/test-elements.R (hunk 2 above)
Comment (klausfrieler) [reply R/test-elements.R:1096]: I think yes, as this caused an error in some circumstances, if I remember well. Unfortunately, I cannot recall the details as this was added quite some time ago now.

File: R/i18n.R (hunk 1 above)
Comment (klausfrieler) [reply R/i18n.R:238]: Sure thing.

**Comment by pmcharrison** [comment]
Thanks! Can you merge once more?

File: R/test-elements.R (hunk 2 above)
Comment (pmcharrison) [reply R/test-elements.R:1096]: Hmm, ok. I think I prefer the original behaviour -- no responses should yield an empty vector, not a vector with an empty string.

**Comment by pmcharrison** [comment]
OK, I think I will just move forward and incorporate the specific change regarding the markdown function. It is a simple one! Thanks for finding it

**Comment by pmcharrison** [comment]
I now merged the relevant fix in my own branch. Should work now. Thanks Klaus!

**Pull request closed without merging**